File: src/textbooks/ctci/review/ds/stack.py
"""

Stack

"""


class Stack(object):

    def __init__(self):
        self.items = []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        return self.items.pop()

    def peek(self):
        return self.items[-1]

    def is_empty(self):
        return self.items == []

    def size(self):
        return len(self.items)

    def __str__(self):
        return ', '.join(str(i) for i in self.items)



if __name__ == '__main__':
    s = Stack()
    for i in range(1, 11):
        s.push(i)
    s.pop()
    s.pop()
    print(s)

File: src/textbooks/ctci/review/ruby/ds/stack.rb
class Stack

  def initialize
    @items = []
  end

  def push(value)
    @items << value
  end

  def pop
    return @items.pop()
  end

  def peek
    return @items[-1]
  end

  def size
    return @items.length
  end

  def is_empty
    return @items.empty?
  end

end



stack = Stack.new()
puts stack.is_empty()
stack.push(4)
stack.push(8)
stack.push(15)
stack.pop()
puts stack.size()
puts stack.is_empty()
puts stack.inspect
